### PR TITLE
Add Nuage manager and event catcher specs

### PR DESCRIPTION
**Pull request by gberginc**
With this patch we are adding specs to test the Nuage network manager,
event catcher and event parser.

### CODE REVIEW

**Comment by gberginc** [comment]
@miq-bot assign @juliancheal 

This PR is now much simpler as we have the other ones merged. There is one issue that I would like to explicitly point out.

[This require](https://github.com/ManageIQ/manageiq-providers-nuage/pull/10/files#diff-93830fa29d616f7c87903d08b5b1b29aR6). As you remember from yesterday we moved these `require`s into methods to prevent breaking the code. This worked for the parser itself, however it causes problems in the specs, because of [this](https://github.com/ManageIQ/manageiq-providers-nuage/blob/master/app/models/manageiq/providers/nuage/network_manager/event_catcher/messaging_handler.rb#L1) -- the implementation extends a class from Qpid. Since spec is ran in isolation, I need to introduce the mandatory require. Let me know if `spec_helper` is the correct place.

@miq-bot add_label test, events

File: spec/models/manageiq/providers/nuage/network_manager/event_catcher/alarm_delete.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
```
Comment (Fryguy) [inline spec/models/manageiq/providers/nuage/network_manager/event_catcher/alarm_delete.json]: Minor, but you might want to move this into a directory named `spec/models/manageiq/providers/nuage/network_manager/event_catcher/data` (we use data as the name in other places).  This way if you add more data files, they are contained.

File: spec/models/manageiq/providers/nuage/network_manager/event_parser_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+describe ManageIQ::Providers::Nuage::NetworkManager::EventParser do
+  context ".event_to_hash" do
+    it "parses vm_ems_ref into event" do
+      ems = FactoryGirl.create(:ems_nuage_network)
+      message = JSON.parse(File.read(File.join(File.dirname(__FILE__), "/event_catcher/alarm_delete.json")))
```
Comment (Fryguy) [inline spec/models/manageiq/providers/nuage/network_manager/event_parser_spec.rb]: Prefer `__dir__` over `File.dirname(__FILE__)`

File: spec/models/manageiq/providers/nuage/network_manager/event_catcher/alarm_delete.json (hunk 1 above)
Comment (Fryguy) [reply spec/models/manageiq/providers/nuage/network_manager/event_catcher/alarm_delete.json]: Oh nvm...I thought the specs files and data would be mixed together (which still *might* happen if you create a new thing namespaced under event_catcher), but this *is* the data directory, so this is fine.

File: spec/models/manageiq/providers/nuage/network_manager_spec.rb [hunk 3]
```diff
@@ -21,5 +38,98 @@
         @ems.verify_credentials(:unsupported)
       end.to raise_error(MiqException::MiqInvalidCredentialsError)
     end
+
+    context 'AMQP connection' do
+      before :each do
```
Comment (Fryguy) [inline spec/models/manageiq/providers/nuage/network_manager_spec.rb]: :each is redundant, so you can remove it.

File: spec/models/manageiq/providers/nuage/network_manager_spec.rb [hunk 4]
```diff
@@ -21,5 +38,98 @@
...
+
+      it 'handles connection errors' do
+        allow(@conn).to receive(:run).and_raise(StandardError, 'connection error')
+        expect { @ems.verify_credentials(:amqp) }.to raise_error(StandardError)
+      end
+    end
+  end
+
+  context 'translate_exception' do
+    before :each do
```
Comment (Fryguy) [inline spec/models/manageiq/providers/nuage/network_manager_spec.rb]: Same.

**Comment by gberginc** [comment]
@Fryguy comments addressed in the spec. Regarding `:each` I have now removed all of them, even in parts that were previously not touched by this PR. I want to avoid any future copy-paste stupidities.

**Comment by gberginc** [comment]
@juliancheal All of the comments have been addressed; @blomquisg approved it so I think it's safe to merge it.

**Pull request merged**